LoginAndAclSetup.md

```diff
@@ -0,0 +1,160 @@
+# Setting up optional login and document level access control
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Setting up Azure AD Apps](#setting-up-azure-ad-apps)
+  - [Manual Setup](#manual-setup)
+    - [Server App](#server-app)
+    - [Client App](#client-app)
+- [Optional Scripts](#optional-scripts)
+  - [Azure Data Lake Storage Gen2 Setup](#azure-data-lake-storage-gen2-setup)
+  - [Azure Data Lake Storage Gen2 Prep Docs](#azure-data-lake-storage-gen2-prep-docs)
+  - [Manually managing Document Level Access Control](#manually-managing-document-level-access-control)
+- [Environment Variables Reference](#environment-variables-reference)
+
+This guide demonstrates how to add an optional login and document level access control system to the sample. This system can be used to restrict access to indexed data to specific users based on what [Azure Active Directory (Azure AD) groups](https://learn.microsoft.com/azure/active-directory/fundamentals/how-to-manage-groups) they are a part of, or their [user object id](https://learn.microsoft.com/partner-center/find-ids-and-domain-names#find-the-user-object-id).
+
+![AppLoginArchitecture](./docs/applogincomponents.png)
+
+## Requirements
+
+**IMPORTANT:** In order to add optional login and document level access control, you'll need the following in addition to the normal sample requirements
+
+* **Azure account permissions**: Your Azure account must have [permission to manage applications in Azure AD](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app#prerequisites).
+
+## Setting up Azure AD Apps
+
+Two Azure AD apps must be registered in order to make the optional login and document level access control system work correctly.
+
+### Manual Setup
+
+The following instructions explain how to setup the two apps using the Azure Portal.
+
+#### Setting up the Server App
```

Comment:
I think my AAD Auth PR automates a lot of this. I prefer automation when possible as its a pain to setup app registrations and even more of a pain to debug the process (when an env variable isn't quite right). Let's join forces!

Reply:
We should do automation, agreed. The problem is setting up the API permissions on the apps, we need to fill in some of the body to the API calls with a bunch of manifest settings. and then we need to go back after the app is provisioned and setup the redirect uris correctly

Reply:
I don't think my branch does the manifest setting, but it does include a postprovision hook to setup the redirect URIs. I can look into the manifest setting automation.

Reply:
we spoke offline - let's get this in a follow up pr